File: app/src/main/java/com/sdk/billinglibrary/LocalConfig.java
package com.sdk.billinglibrary;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.annotation.Nullable;

class LocalConfig {

    private static final String PREFERENCES = "billing";
    private static final String KEY_SUBSCRIBED = "subscribed";
    private static final String KEY_IS_FIRST_TIME_BILLING = "first_time";

    private static SharedPreferences preferences;

    static void init(Context context) {
        if (preferences == null)
            preferences = context.getSharedPreferences(PREFERENCES, Context.MODE_PRIVATE);
    }

    static void subscribeLocally(String subId) {
        if (subId != null)
            preferences.edit().putString(KEY_SUBSCRIBED, subId).apply();
        else
            preferences.edit().remove(KEY_SUBSCRIBED).apply();
    }

    static @Nullable String getCurrentSubscription() {
        return preferences.getString(KEY_SUBSCRIBED, null);
    }

    static boolean isFirstTimeBilling() {
        return !preferences.contains(KEY_IS_FIRST_TIME_BILLING);
    }

    static void didFirstBilling() {
        preferences.edit().putBoolean(KEY_IS_FIRST_TIME_BILLING, true).apply();
    }
}

File: app/src/main/java/com/sdk/billinglibrary/LocalConfig.kt
package com.sdk.billinglibrary

import android.content.Context
import android.content.SharedPreferences
import androidx.core.content.edit

object LocalConfig {

    private const val PREFERENCES = "billing"
    private const val KEY_SUBSCRIBED = "subscribed"
    private const val KEY_IS_FIRST_TIME_BILLING = "first_time"

    private lateinit var preferences: SharedPreferences

    fun init(context: Context) {
        preferences = context.getSharedPreferences(PREFERENCES, Context.MODE_PRIVATE)
    }

    fun subscribeLocally(subId: String?) {
        if (subId != null)
            preferences.edit { putString(KEY_SUBSCRIBED, subId) }
        else
            preferences.edit { remove(KEY_SUBSCRIBED) }
    }

    fun getCurrentSubscription(): String? {
        return preferences.getString(KEY_SUBSCRIBED, null)
    }

    fun isFirstTimeBilling(): Boolean {
        return !preferences.contains(KEY_IS_FIRST_TIME_BILLING)
    }

    fun didFirstBilling() {
        preferences.edit { putBoolean(KEY_IS_FIRST_TIME_BILLING, true) }
    }
}
